app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaMigration.java

```diff
@@ -0,0 +1,54 @@
+package com.appsmith.server.migrations;
+
+import com.appsmith.server.domains.ApplicationJson;
+import com.appsmith.server.exceptions.AppsmithError;
+import com.appsmith.server.exceptions.AppsmithException;
+
+public class JsonSchemaMigration {
+    private static Boolean checkCompatibility(ApplicationJson applicationJson) {
+        return (applicationJson.getClientSchemaVersion() <= JsonSchemaVersions.clientVersion)
+                && (applicationJson.getServerSchemaVersion() <= JsonSchemaVersions.serverVersion);
+    }
+
+    public static ApplicationJson migrateApplicationToLatestSchema(ApplicationJson applicationJson) {
+        // Check if the schema versions are available and set to initial version if not present
+        Integer serverSchemaVersion = applicationJson.getServerSchemaVersion() == null ? 0 : applicationJson.getServerSchemaVersion();
+        Integer clientSchemaVersion = applicationJson.getClientSchemaVersion() == null ? 0 : applicationJson.getClientSchemaVersion();
+
+        applicationJson.setClientSchemaVersion(clientSchemaVersion);
+        applicationJson.setServerSchemaVersion(serverSchemaVersion);
+        if (Boolean.FALSE.equals(checkCompatibility(applicationJson))) {
```

Comment:
If the return type is changed to `boolean`, this can just be:

```suggestion
        if (!checkCompatibility(applicationJson)) {
```